### DISCUSSION TITLE
[ENHANCEMENT] Repository for the byzx files

### DISCUSSION THREAD

**Discussion by ghost** [neanes/neanes#460] [Ideas]
I want to ask you to make a repository for the **byzx** files
It would be also wonderful if you put on the site the possibility for anyone to upload **byzx** files, because most of those who will use **neanes**, doesn't know what **github** is and how to use it.
I think that in this way it will help many people and also make **neanes** much more popular.

**Comment by danielgarthur** [comment]
I like the idea of allowing people to share their files, but I'm not sure how to best monitor file content. There's probably many more considerations, but two that come to mind are the following.

- How to prevent scammers/bots from uploading inappropriate or dangerous files?
- How to deal with copyright issues? Many English texts are copyrighted and require permission from the author to publish in any form. I suspect other non-English translations also have similar copyright issues.

**Comment by ghost** [comment]
For uploading you can restrict file format that can be uploaded to **byzx** and use something like **reCAPTCHA**.
I did not realize that there is this **Copyright** restriction even for Church texts.
Wouldn't be enough to have a way for anybody to complain if a **byzx** file has **Copyright** issue and only then to be removed?
I know that there are many programs with which you can write scores for **Byzantine Chant**, but [Melodos](https://www.melodos.com/) is by far the most popular and also the best.
**neanes** is much more straightforward than **Melodos** and easy to use, I hope that in time it will replace it.
There are also a lot of **mel** files that have been created over the years, even I have around 200.
**mel** is not plain text file like **byzx**, so I don't think a method can be found to convert ***.mel** to ***.byzx**

**Comment by basil** [comment]
Anyone can create and curate a repository of scores on GitHub. If copyright infringement occurs, a complaint can be submitted to GitHub Trust & Safety. GitHub Trust & Safety takes these complaints seriously. I would not recommend exposing yourself to this risk. (I have experience with this in another open source project I maintain.)

The more interesting topic from my perspective is how to facilitate collaboration and improvement of compositions. Suppose copyright is not an issue, e.g. in the case of ancient Greek texts in the public domain or a private repository making legal use of licensed copyrighted texts. If I see something in a composition that could be improved (e.g., some awkward rhythm that could be smoother with the addition of a gorgon), I would like to be able to submit a pull request to improve it. Similarly, I would like to be able draft a composition, open a draft pull request, and request a review from other composers.

For this to work, the representation of the content has to make some sense in the GitHub UI, even if it is not rendered. For example, Markdown or AsciiDoc can be reasonably reviewed in the GitHub UI without rendering. (Certain AsciiDoc conventions, like using one sentence per line, improve reviewability.)

**Comment by basil** [comment]
I started putting some of my compositions online at https://github.com/basil/byzantine-music using a Byzantine Music as Code (BMaC) approach. If you think something can be improved, pull requests are welcome.

My long-term vision is to have some sort of build system that will compile the uncompressed Neanes files and generate a web site of HTML and PDF files based on the folder structure of `.byzx` files.
